### ISSUE TITLE
Censorize : adding noise turns the image black

### ISSUE THREAD

**Issue by chhil** [darktable-org/darktable#7678]
Censorize : Simply changing the noise slider turns the image black. Changing the value does not change the black portion. 
The module has to be reset.

DT master , opencl off. WSL Ubuntu 20.04

![image](https://user-images.githubusercontent.com/1119439/103549610-a943ff00-4ecd-11eb-9135-f50bc16282b9.png)

**Comment by chhil** [comment]
History stack when I tried it.

![image](https://user-images.githubusercontent.com/1119439/103550217-954ccd00-4ece-11eb-820e-7d09b96e1a25.png)

**Comment by johnny-bit** [comment]
We'd need xmp file that we can try to replicate the issue... Also you could run dt with nans debug enabled to see if there are any NANs produced that could explode things...

**Comment by chhil** [comment]
I tried it with a different image and got the problem....

![censor](https://user-images.githubusercontent.com/1119439/104281834-1eb65d80-54d4-11eb-9422-fb982db58f0f.gif)

Then when I restarted dt with -d nan, I could not reproduce it on the same image.
Irrespective I have attached the debug output.
[debugNAN.txt](https://github.com/darktable-org/darktable/files/5800447/debugNAN.txt)

**Comment by chhil** [comment]
I tried it on a different image and got black boxes. Also note the preview top left has only one black box on the right.

Ignore the blurry image, it was exported using lesser colors to reduce file size. 
![censor2](https://user-images.githubusercontent.com/1119439/104283785-10b60c00-54d7-11eb-8b2a-d80d2f93ef7a.gif)
 

Then when I restarted dt with -d nan, I could not reproduce it on the same image.
Irrespective I have attached the debug output.

**Comment by Nilvus** [comment]
I can reproduce too with that module but only if I use only noise slider with so all other to 0. If I set one or the other slider to a value different than 0, no more black boxes. @aurelienpierre: is it normal as this module probably need at least one blur to be applied? If yes, maybe a warning is needed there. If not, a bug to fix.

**Comment by chhil** [comment]
Still happens today

3.5.0+3.5.0+1150~gbff50e6e1-dirty

Its dirty, as I have manually applied a patch for a seg fault caused by Censorize.

![image](https://user-images.githubusercontent.com/1119439/108651466-cff2df00-74e7-11eb-856b-e1ea60a52b02.png)


Then I created a dup using original. Went step by step adding the modules. Though I added censorize early as seen in the history and image was normal. Then I added contrast equalizer (preset denoise+sharpen) and I get the black boxes.

Contrast equalizer added.
![image](https://user-images.githubusercontent.com/1119439/108651977-f402f000-74e8-11eb-8089-f8ac1f1087c1.png)

If I toggle contrast equalizer on/off, I get the black boxes appear and disappear.

The strange thing though is that I can turn the boxes on/off  by disabling color calibration too.

When I disable haze, some of the black boxes disappear.

I shared the image and the xmp.

https://www.dropbox.com/s/zmg7oa0cp5rs0fb/Issue%207678.7z?dl=0

Haze diabled.
![image](https://user-images.githubusercontent.com/1119439/108652149-7390bf00-74e9-11eb-8fa1-e04ad671ab3d.png)
